### ISSUE TITLE
Input type checkbox doesn't get updated when submitting on change

### ISSUE THREAD

**Issue by apieceofbart** [jaredpalmer/formik#1963]
### Current Behavior

I have a simple form with one checkbox that is unchecked by default. On the form I am using `onChange={formik.handleSubmit}` because I want to save form on change without a submit button.
When I click on the checkbox I can see that the internal value has changed but the checkbox is still unchecked. I have to click it again to make it checked but then internal value of this field does not change.
I can fix that by adding `key={values.toggle.toString()}` on the input to force re-render but of course this is a bad solution

### Expected behavior
I would expect the checkbox to work as it should and follow the internal value

### Reproducible example

Check codesandbox: https://codesandbox.io/s/formik-checkbox-issue-ew65e

### Your environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Formik           | 1.5.8
| React            | 16.8.6
| TypeScript       |
| Browser          |
| npm/Yarn         |
| Operating System |

**Comment by jaredpalmer** [comment]
The reason for the weird behavior is that `handleSubmit` calls `event.preventDefault` and `event.stopPropagation` internally because it is expecting a submit event (hence the name _handleSubmit_). When this you pass it to `onChange` is thus messes with the synthetic event bubbling. A better and safer way to do autosave is to use either `useEffect(() => {}, [formik.values])` or a child class component that uses `componentDidUpdate(prevProps, prevState)`.

**Comment by jaredpalmer** [comment]
Here is an example of just using `<form onChange={() => submitForm()}` instead of `<form onChange={handleSubmit}>` 

https://codesandbox.io/s/formik-checkbox-issue-3rktu

**Comment by apieceofbart** [comment]
@jaredpalmer  Thank you very much for quick reply! Meanwhile I noticed that most of my problems come from doing the autosave wrong and I found a solution in one of the issues.
(I don't think your sandbox example works as expected, it doesn't seem to log the submitted values)

Maybe we could add an example to readme with a proper way of handling autosave in formik? I can help with that (after agreeing what it should look like).
